### DISCUSSION TITLE
fieldArray items not being stored in array

### DISCUSSION THREAD

**Discussion by BenSeward** [react-hook-form/react-hook-form#1915] [Q&A] [answered]
Hi everyone! I was hoping someone might be able to help with this as I've been racking my brain on this for a few days and can't figure it out. 

I have a large form that I was to break up into sub-components. However when trying to use a fieldArray the different fields aren't stored in their own array, they're stored at the root of the form object. 

**This is what the current form data looks like:**
```
{
  firstname: "testy"
  lastname: "testington"
  gender: "male"
  exercises[0].name: "exercise 1"
  exercises[1].name: "test 2"
}
```

**This is what I need the form data to look like:**
```
{
  firstname: "testy"
  lastname: "testington"
  gender: "male"
  exercises: [{name: "exercise 1"}, {name: "exercise 2"}]
}
```

I've attached a codesandbox for reference, I'd really appreciate the help and thank you! :-)

[https://codesandbox.io/s/confident-cartwright-r1z5u](https://codesandbox.io/s/confident-cartwright-r1z5u)

**Comment by bluebill1049** [comment, marked as answer]
getValue({ nest: true }) we are fixing this on V6 FYI

getValue() in v6 will return nested value by default.

**Reply by BenSeward** [reply]
This is great thanks Bill!
